src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1830,7 +1830,7 @@ bool Arguments::check_vm_args_consistency() {
   }
 
   // Valhalla missing LM_LIGHTWEIGHT support just now
-  if (EnableValhalla && LockingMode != LM_LEGACY) {
+  if (EnableValhalla && LockingMode == LM_LIGHTWEIGHT ) {
```

Comment:
Nit: remove the space after LIGHTWEIGHT.

This should work.  If lightweight locking is specified on the command line, you probably should have a warning, but I assume this is temporary until it is supported.